neps/nep-0509.md

```diff
@@ -473,34 +473,34 @@ This makes chunk validator selection algorithm correctness critical for the secu
-TODO
+The only real alternative that was considered is the original nightshade proposal. The full overview of the differences can be found in the revised nightshade whitepaper at https://near.org/papers/nightshade. 
 
 ## Future possibilities
 
 * Integration with ZK allowing to get rid of large state witness distribution. If we treat state witness as a proof and ZK-ify it, anyone can validate that state witness indeed proves the new chunk header with much lower effort. Complexity of actual proof generation and computation indeed increases, but it can be distributed among chunk producers, and we can have separate concept of finality while allowing generic users to query optimistic chunks.
-* TODO [Describe any natural extensions and evolutions to the NEP proposal, and how they would impact the project. Use this section as a tool to help fully consider all possible interactions with the project in your proposal. This is also a good place to "dump ideas"; if they are out of scope for the NEP but otherwise related. Note that having something written down in the future-possibilities section is not a reason to accept the current or a future NEP. Such notes should be in the section on motivation or rationale in this or subsequent NEPs. The section merely provides additional information.]
+* Integration with resharding to further increase the number of shards and the total throughput. 
+* The sharding of non-validating nodes and services. There are a number of services that may benefit from tracking only a subset of shards. Some examples include the RPC, archival and read-RPC nodes. 
 
 ## Consequences
 
-[This section describes the consequences, after applying the decision. All consequences should be summarized here, not just the "positive" ones. Record any concerns raised throughout the NEP discussion.]
-TODO
-
 ### Positive
 
-- The validator nodes will only need to track one shard. 
-- 
+- The validator nodes will need to track at most one shard. 
+- The state will be held in memory making the chunk application much faster.
+- Thanks to the above, in the future, it will be possible to reduce the gas costs and by doing so increase the throughput of the system. 
 
 ### Neutral
 
-- n1
+- The current approach to resharding will need to be revised to support generating state witness. 
+- The security assumptions will change. The responsibility will be moved from block producers to chunk validators and the security will become probabilistic. 
 
 ### Negative
 
-- n1
+- The hardware requirements will increase. In particular the network bandwidth and memory requirements will be higher. 
+- Additional limits will be put on the size of transactions, receipts and, more generally, cross shard communication.
+- The dependency on cloud state sync will increase the centralization of the blockchain. This will be resolved separately by the decentralized state sync. 
```

Comment:
should we specify what it starts having a dependency on state sync?

Reply:
Well, it's mentioned above on explanation of validator roles.